src/Demo.Infrastructure/Repositories/ProductRepository.cs

```diff
@@ -48,17 +49,17 @@ public async Task<IEnumerable<Product>> GetAll()
             //return await QueryAsync<Product>("SELECT ProductId, Name, QuantityPerUnit, UnitPrice, UnitsInStock, UnitsOnOrder, ReorderLevel, Discontinued, CategoryId FROM PRODUCT");
 
             // Option 6: Standard EFCore Way
-            return await _demoDbContext.Products
-                .Include(x => x.Category)
-                .ToListAsync();
-        }
 
-        public async Task<IEnumerable<Product>> GetByName(string productName)
-        {
-            return await _demoDbContext.Products
-                .Where(x => x.Name == productName)
+            var pagedData = await _demoDbContext.Products
                 .Include(x => x.Category)
+                .Skip((queryStringParameters.PageNumber - 1) * queryStringParameters.PageSize)
+                .Take(queryStringParameters.PageSize)
                 .ToListAsync();
+
+            var totalRecords = await _demoDbContext.Products.CountAsync();
```

Comment:
Should we make this behavior configurable i.e. to allow clients denying the pagination data and in that case it doesn't need to count the total records

Reply:
@barotbm This particular endpoint will always serve pagination data whether Client specify pagination or not. If Client doesn't specify pagination then we are still returning data with default pagination parameter (Current Page = 1 & page Size = 10), so we need to return that totalRecords in that case as well.

Can you please provide an example of your scenario?

Reply:
Example: From client application, lets say I got the necessary pagination metadata in my first call but the second call onwards I don't want to receive the pagination info because I am caching it on the UI. So in the next call I want to pass a parameter not to make any database queries to fetch the total records.

Reply:
@barotbm Done. Added IncludeTotalCount property with default as false.

Reply:
@barotbm please review and if it seems ok then merge PR.